File: solutions/lua/resistor-color-trio/1/resistor-color-trio.lua
return {
  decode = function(c1, c2, c3)
    local unit=0
    local value=0
    local res={}
    res['black']=0
    res['brown']=1
    res['red']=2
    res['orange']=3
    res['yellow']=4
    res['green']=5
    res['blue']=6
    res['violet']=7
    res['grey']=8
    res['white']=9

    value=res[c1]*10 + res[c2]

    if res[c3] == 0 then 
      unit="ohms"
    elseif res[c3] == 1 then 
      unit="ohms"
      value = value * 10
    elseif res[c3] == 2 then 
      unit="kiloohms"
      value = value / 10
    elseif res[c3] == 3 then 
      unit = "kiloohms"
    elseif res[c3] == 4 then
      unit = "kiloohms"
      value = value * 10
    end 
    return value,unit
  end
}

--print(decode('red','red','red'))

File: solutions/c/resistor-color-trio/1/src/resistor_color_trio.c
#include "resistor_color_trio.h"

resistor_value_t color_code(resistor_band_t* test)
{
    resistor_value_t var01;

//    if(test[1]!=BLACK)
        var01.value=(test[0]*10) + test[1];
//    else
//        var01.value=test[0];

    switch (test[2])
    {
    case BLACK:
        var01.unit=OHMS;
        break;
    case BROWN:
        var01.value*=10;
        var01.unit=OHMS;
        break;
    case RED:
        var01.unit=KILOOHMS;
        var01.value/=10;
        break;
    case ORANGE:
        var01.unit=KILOOHMS;
        break;
    case YELLOW:
        var01.value*=10;
        var01.unit=KILOOHMS;
        break;
    default:
        break;
    }
    return var01;
}